tests/slack_sdk/socket_mode/test_interactions_builtin.py

```diff
@@ -126,7 +126,44 @@ def socket_mode_request_handler(
                 self.logger.info(f"Passed with buffer size: {buffer_size}")
 
         finally:
+            client.close()
             self.server.stop()
             self.server.close()
 
         self.logger.info(f"Passed with buffer size: {buffer_size_list}")
+
+    def test_sending_messages(self):
+        if is_ci_unstable_test_skip_enabled():
+            return
+        t = Thread(target=start_socket_mode_server(self, 3011))
+        t.daemon = True
+        t.start()
+        time.sleep(2)  # wait for the server
+
+        try:
+            self.reset_sever_state()
+            client = SocketModeClient(
+                app_token="xapp-A111-222-xyz",
+                web_client=self.web_client,
+                auto_reconnect_enabled=False,
+                trace_enabled=True,
+            )
+            client.wss_uri = "ws://0.0.0.0:3011/link"
+            client.connect()
+            time.sleep(1)  # wait for the connection
+            client.send_message("foo")
+
+            client.disconnect()
+            time.sleep(1)  # wait for the connection
+            try:
+                client.send_message("foo")
+            except SlackClientNotConnectedError as _:
+                pass
```

Comment:
Ideally we should verify if there are expected logs in this test but I manually checked `logs/pytest.log`.

Reply:
It is hard for me to understand the intention of this test. The name of the test, "test sending messages", could be interpreted as validating a variety of behaviours that make up the overall message sending functionality. In context of this PR, I assume that this test checks for the single-retry-when-underlying-WS-connection-is-re-established behaviour, but perhaps a week or two from now I will no longer remember this (my memory is not what it used to be 👴 ).

Two questions:

- could we rename the test to be more specific as to what we are validating?
- can we add assertions to the test that make explicit what we expect to happen? I assume that in this test we want  to trigger the `SlackClientNotConnectedError`, but because we `pass` on it, if the error is not raised, then the test passes. So in effect this test may - or may not - validate behaviour, in that if the exception is raised, we `pass`, but if it is not, we implicitly `pass`.

Reply:
>could we rename the test to be more specific as to what we are validating?

You are definitely right here. Perhaps, a better test name would be `test_send_message_while_disconnection`

>if the error is not raised, then the test passes. 

Good point. we can add an explicit failure for the case where the exception is not propagated. I will update the tests and the typo in comments later on.

Reply:
Thank you!